Gems/ROS2/Code/Source/Gripper/VacuumGripper.cpp

```diff
@@ -0,0 +1,336 @@
+        jointConfig.m_childLocalPosition = childTransformParent.GetTranslation();
+        jointConfig.m_startSimulationEnabled = true;
+
+        // create new joint
+        m_vaccumJoint =
+            sceneInterface->AddJoint(defaultSceneHandle, &jointConfig, m_gripperEffectorBodyHandle, grippedRigidBody->m_bodyHandle);
+
+        return true;
+    }
+
+    void VacuumGripper::ReleaseGrippedObject()
+    {
+        m_gripping = false;
+        if (m_vaccumJoint == AzPhysics::InvalidJointHandle)
+        {
+            return;
+        }
+        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AzPhysics::SceneHandle defaultSceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        AZ_Assert(defaultSceneHandle != AzPhysics::InvalidSceneHandle, "Invalid default physics scene handle");
+        // Wake up the body to prevent it from not moving after release
+        AzPhysics::RigidBody* grippedRigidBody = nullptr;
+        Physics::RigidBodyRequestBus::EventResult(grippedRigidBody, m_grippedObjectInEffector, &Physics::RigidBodyRequests::GetRigidBody);
+        if (grippedRigidBody)
+        {
+            grippedRigidBody->ForceAwake();
+        }
+        sceneInterface->RemoveJoint(defaultSceneHandle, m_vaccumJoint);
+        m_vaccumJoint = AzPhysics::InvalidJointHandle;
+    }
+
+    void VacuumGripper::OnImGuiUpdate()
+    {
+        ImGui::Begin("VacuumGripperDebugger");
+
+        AZStd::string grippedObjectInEffectorName;
+        AZ::ComponentApplicationBus::BroadcastResult(
+            grippedObjectInEffectorName, &AZ::ComponentApplicationRequests::GetEntityName, m_grippedObjectInEffector);
+
+        ImGui::Text("Grippable object : %s", grippedObjectInEffectorName.c_str());
+        ImGui::Text("Vacuum joint created : %d ", m_vaccumJoint != AzPhysics::InvalidJointHandle);
+
+        ImGui::Checkbox("Gripping", &m_gripping);
+
+        if (ImGui::Button("Grip Command "))
+        {
+            m_gripping = true;
+        }
+
+        if (ImGui::Button("Release Command"))
+        {
+            ReleaseGrippedObject();
+        }
+        ImGui::End();
+    }
+
+    AZ::Outcome<void, AZStd::string> VacuumGripper::GripperCommand(float position, float maxEffort)
+    {
+        AZ_Printf("VacuumGripper", "GripperCommand %f", position);
+        if (position > 0)
+        {
+            m_gripping = true;
+        }
+        else
+        {
+            ReleaseGrippedObject();
+        }
+        return AZ::Success();
+    }
+
+    AZ::Outcome<void, AZStd::string> VacuumGripper::CancelGripperCommand()
+    {
+        ReleaseGrippedObject();
+        return AZ::Success();
+    }
+
+    float VacuumGripper::GetGripperPosition() const
+    {
+        return m_vaccumJoint == AzPhysics::InvalidJointHandle ? 0.0f : 1.0f;
+    }
+
+    float VacuumGripper::GetGripperEffort() const
+    {
+        return m_vaccumJoint == AzPhysics::InvalidJointHandle ? 0.0f : 1.0f;
+    }
```

Comment:
These functions have the same implementation.
Should that be the case?

Reply:
It is an ultra simple gripper, for a more advanced one, implementation will be different.